webpack.config.js

```diff
@@ -31,9 +31,32 @@ const requestToHandle = ( request ) => {
 
 	return wcHandleMap[ request ];
 };
+const exceptSVGRule = ( rule ) => {
+	return ! rule.test.toString().match( /svg/i );
+};
 
 const webpackConfig = {
 	...defaultConfig,
+	module: {
+		...defaultConfig.module,
+		// Expose image assets as files.
+		// In Webpack 5 we would use Asset Modules and `asset/resource`
+		rules: [
+			// Remove `@wordpress/` rules for SVGs.
+			...defaultConfig.module.rules.filter( exceptSVGRule ),
+			{
+				test: /\.(svg|png|jpe?g|gif)$/i,
+				use: {
+					loader: 'file-loader',
+					options: {
+						name: 'images/[path]/[contenthash].[name].[ext]',
```

Comment:
> If we (accidentally or intentionally) have duplicate SVG files under `js/src`, it'd only emit one SVG to `js/build/images` due to they have the same content hash. Thus, the archive zip file size would not increase the duplicate parts.

I was thinking about it as saving package save and de-duplicating is a valuable gain. But to achieve that we would have to either the same filename or reduce it to `'images/[contenthash].[ext]'`. Which then impairs debugging, development, and extremely niche user experience if someone happens to download those files. The last one could be ignored for our plugin, but I think a debugging case could be valuable.

Imagine a case, when we have a merchant reporting some issue to a Happiness Engineer, and HE discovered a `404 /images/afu76poo0a0602bca20d01a0ff048516339987a8ad1` in logs. To resolve the file to even know whether the 404 is something related to the issue or not, they need either guess, run some tooling, run the build for the exact revision of the plugin, etc.

If the log looks like `404 /images/js/src/components/account-card/afu76poo0a0.gogole-g-logo.svg`, they could more easily judge
- "Ok, this is not related to our issue of Automatewoo not triggering…"
- "Ah, so the image file is not there, could you check it in the file system", or "Maybe you happen to block URLs with bad words, and `poo` got caught"

> It doesn't affect much, but the final URL could be shorter, and no need to have the full subpath under `js/build/images`.


Personally, I considered full path a DevX as it allows a developer to match 1:1 the files used on production build to the file in their source.

To simplify/shorten the file path we could try to remove `/js/src` from there. So to make `/js/src/components/account-card/index.js` & `/js/src/components/account-card/gogole-g-logo.svg`, become `/js/build/index.js` & `/js/build/images/components/account-card/gogole-g-logo.svg` (instead of `/js/build/images/js/src/components/account-card/gogole-g-logo.svg`) .
But I'm not sure how easy would it be.

Reply:
To me, it looks like a trade-off between
- de-duplication, vs.
- debugging experience

WDYT?

Reply:
Thanks for sharing your thoughts. I believe both of these scenarios will happen very rarely. But once if any of them occur, definitely the debugging experience brings more benefit.